config/apigroup_config.go

```diff
@@ -0,0 +1,712 @@
+		"azurerm_data_factory_linked_service_data_lake_storage_gen2":                     "datafactory",
+		"azurerm_data_factory_integration_runtime_azure":                                 "datafactory",
+		"azurerm_data_lake_analytics_account":                                            "datalakeanalytics",
+		"azurerm_data_lake_analytics_firewall_rule":                                      "datalakeanalytics",
+		"azurerm_data_lake_store_virtual_network_rule":                                   "datalakestore",
+		"azurerm_data_lake_store":                                                        "datalakestore",
+		"azurerm_data_lake_store_firewall_rule":                                          "datalakestore",
+		"azurerm_database_migration_project":                                             "datamigration",
+		"azurerm_database_migration_service":                                             "datamigration",
+		"azurerm_data_protection_backup_instance_blob_storage":                           "dataprotection",
+		"azurerm_data_protection_backup_policy_postgresql":                               "dataprotection",
+		"azurerm_data_protection_backup_policy_blob_storage":                             "dataprotection",
+		"azurerm_data_protection_backup_policy_disk":                                     "dataprotection",
+		"azurerm_data_protection_backup_instance_disk":                                   "dataprotection",
+		"azurerm_data_protection_backup_vault":                                           "dataprotection",
+		"azurerm_data_protection_backup_instance_postgresql":                             "dataprotection",
+		"azurerm_data_share":                                                             "datashare",
+		"azurerm_data_share_account":                                                     "datashare",
+		"azurerm_data_share_dataset_data_lake_gen1":                                      "datashare",
+		"azurerm_data_share_dataset_kusto_database":                                      "datashare",
+		"azurerm_data_share_dataset_blob_storage":                                        "datashare",
+		"azurerm_data_share_dataset_kusto_cluster":                                       "datashare",
+		"azurerm_data_share_dataset_data_lake_gen2":                                      "datashare",
+		"azurerm_mariadb_database":                                                       "dbformariadb",
+		"azurerm_mariadb_configuration":                                                  "dbformariadb",
+		"azurerm_mariadb_firewall_rule":                                                  "dbformariadb",
+		"azurerm_mariadb_virtual_network_rule":                                           "dbformariadb",
+		"azurerm_mariadb_server":                                                         "dbformariadb",
+		"azurerm_mysql_configuration":                                                    "dbformysql",
```

Comment:
This group seems to be how it is grouped in TF documentation page.

<img width="323" alt="Screen Shot 2022-01-03 at 16 06 00" src="https://user-images.githubusercontent.com/9900707/147933871-d15d55aa-7d06-410f-8980-c82e0bcf918b.png">

Currently `azurerm_mysql_server` goes under `dbformysql` group and `azurerm_mssql_server` goes under `sql` ...

I believe having all under `database` group would be more intutive.

Reply:
Similar to the reasoning stated above, `mysql` related resources use `Microsoft.DBforMySQL` as their native provider name. Because majority (if not all) Azure REST resources have the native provider name in their resource paths, these native provider names make consistent and easily predictable group names for our resources. Also it makes maintenance easier as we are utilizing Microsoft's grouping decisions, and we have been able to scrape them from the Terraform registry (otherwise it might have been a tedious task to employ native provider names).

While I also find grouping all database related resources under the `database` group reasonable and easy to implement by changing the corresponding map values, sticking to the new convention mentioned above has its own merits.

Reply:
As we discussed offline yesterday, I am fine with both approach as long as we are consistent. I just feel like this approach would require more work to finalize Kind names since we no longer extract group from resource name and leave the rest as kind name, but if you think otherwise, I am fine with that.

Reply:
@ulucinar would you mind sharing a list of all resource group/kind mapping as done in the PR description [here](https://github.com/crossplane-contrib/provider-jet-gcp/pull/25).

<img width="499" alt="Screen Shot 2022-01-05 at 15 42 28" src="https://user-images.githubusercontent.com/9900707/148219408-5e8a04a2-dedd-4797-8780-16158ce8fe72.png">

Reply:
Hi @turkenh,
Please see the PR description for a list of kind-group mappings.